doc/source/tutorial/interpolate.rst

```diff
@@ -10,38 +10,494 @@ Interpolation (:mod:`scipy.interpolate`)
+capabilities: it can construct splines of various degrees (via the optional
+argument ``k``) and predefined knots (via the optional argument ``t``). 
+
+A b-spline of degree ``k`` is defined by its knots and coefficients. The knots
+are availble as the ``t`` attribute of a `BSpline` instance:
+
+    >>> print(bspl.t)
+    [0.  0.  0.  0.        0.5  0.75  1.        1.5  1.5  1.5  1.5 ]
+    >>> print(x)
+    [            0.  0.25  0.5  0.75  1.  1.25  1.5 ]
+
+We see that the knot vector by default is constructed from the input
+array ``x``: first, it is made :math:`(k+1)` -regular (it has ``k+1``
+repeated knots appended and prepended); then, the second and
+second-to-last points of the input array are removed---this is the so-called
+*not-a-knot* boundary condition. 
+
+In general, an interpolating spline of degree ``k`` needs
+``len(t) - len(x) - k - 1`` boundary conditions. For cubic splines with
+``(k+1)``-regular knot arrays this means two boundary conditions---or
+removing two values from the ``x`` array. Various boundary conditions can be
+requested using the optional ``bc_type`` argument of `make_interp_spline`.
+
+The b-spline coefficients are accessed via the ``c`` attribute of a `BSpline`
+object:
+
+    >>> len(bspl.c)
+    7
+
+The convention is that for ``len(t)`` knots there are ``len(t) - k - 1``
+coefficients. Some routines (see the :ref:`Smoothing splines section
+<tutorial-interpolate_fitpack>`) zero-pad the ``c`` arrays so that
+``len(c) == len(t)``. These additional coefficients are ignored for evaluation.
+
+Also note that the coeffients are given in the
+:ref:`b-spline basis <tutorial-interpolate_bspl_basis>`, not the power basis
+of :math:`1, x, \cdots, x^k`.
+
+
+.. _tutorial-interpolate_bspl_basis:
+
+B-spline basis elements
+-----------------------
+
+B-splines are piecewise polynomials, represented as linear combinations of
+*b-spline basis elements* --- which themselves are certain linear combinations
+of usual monomials, :math:`x^m` with :math:`m=0, 1, \dots, k`.
+
+The b-spline basis generally more computationally stable then the power basis
+and is useful for variety of applications which include interpolation, regression
+and curve representation. The main feature is that these basis elements are
+*localized* and equal zero outside of an interval defined by the *knot array*.
+
+Specifically, a b-spline basis element of degree ``k`` (e.g. ``k=3`` for cubics)
+is defined by :math:`k+2` knots and is zero outside of these knots.
+To illustrate, plot a collection of non-zero basis elements on a certain
+interval:
+
+.. plot ::
+
+    >>> k = 3      # cubic splines
+    >>> t = [0., 1.4, 2., 3.1, 5.]   # internal knots
+    >>> t = np.r_[[0]*k, t, [5]*k]   # add boundary knots
+
+    >>> from scipy.interpolate import BSpline
+    >>> import matplotlib.pyplot as plt
+    >>> for j in [-2, -1, 0, 1, 2]:
+    ...     a, b = t[k+j], t[-k+j-1]
```

Comment:
"and is zero outside of these knots"
Maybe this point would be illustrated better if `a = 0` and `b = 5` (fixed)? Right now, I can't see what the values are outside the interval.

Reply:
Sadly, these things extrapolate by default, so no, it might be even more confusing. I guess I'll need to say it in the text.